asterius/src/Asterius/Backends/Binaryen.hs

```diff
@@ -207,164 +217,219 @@ marshalFunctionType FunctionType {..} = flip runContT pure $ do
     rt <- c_BinaryenTypeCreate rts (fromIntegral rtl)
     pure (pt, rt)
 
-marshalExpression ::
-  Bool ->
-  M.Map AsteriusEntitySymbol Int64 ->
-  BinaryenModuleRef ->
-  Expression ->
-  IO BinaryenExpressionRef
-marshalExpression tail_calls sym_map m e = flip runContT pure $ case e of
+-- TODO: there is a similar definition in Asterius.Backends.WasmToolkit. Maybe
+-- we should just move shared stuff into Asterius.Types (or another shared file).
+type SymbolMap = M.Map AsteriusEntitySymbol Int64
+
+-- | Environment used during marshaling of Asterius' types to Binaryen.
+data MarshalEnv
```

Comment:
Since we're adding a new datatype in this module, I suggest adding the `StrictData` extension here, since we don't use laziness in any field; `StrictData` is a good default to get a bit of speed up and plug in some leaks imo.

Reply:
Good point :+1: